packages/wallets/src/components/Base/ModalStepWrapper/ModalStepWrapper.tsx

```diff
@@ -21,11 +21,14 @@ const ModalStepWrapper: FC<PropsWithChildren<TModalStepWrapperProps>> = ({
 }) => {
     const { hide } = useModal();
     const hasRenderFooter = typeof renderFooter === 'function';
+    const fixedFooter = shouldFixedFooter && hasRenderFooter;
 
     return (
         <div
             className={classNames('wallets-modal-step-wrapper', {
-                'wallets-modal-step-wrapper--fixed': shouldFixedFooter,
+                'wallets-modal-step-wrapper--fixed-footer': fixedFooter && !shouldHideHeader,
+                'wallets-modal-step-wrapper--no-header': shouldHideHeader && !fixedFooter,
+                'wallets-modal-step-wrapper--no-header--fixed-footer': shouldHideHeader && fixedFooter,
```

Comment:
minor - typically the patterns is to keep modifiers independant and not combine them (otherwise, we can end up with very long class names containing every possible combination and with complex logic to apply proper combinations),